pkgs/development/python-modules/setuptools-rust/setuptools-rust-hook.sh

```diff
@@ -0,0 +1,27 @@
+echo "Sourcing setuptools-rust-hook"
+
+setuptoolsRustSetup() {
+    # This can work only if rustPlatform.cargoSetupHook is also included
+    if ! command -v cargoSetupPostPatchHook >/dev/null; then
+        echo "ERROR: setuptools-rust has to be used alongside with rustPlatform.cargoSetupHook."
+        exit 1
+    fi
+
+    # We can't easily decide which target Python interpreter should be used when
+    # building setuptools-rust. It is the same version as the one we use to
+    # build setuptools-rust for but possibly for a different target platform.
+    # The easiest way is to use Python as provided in the package that uses this
+    # hook.
+    local pypath
+    pypath="$(command -v python)"
+    pypath="${pypath%/bin/python}"
+    export PYO3_CROSS_LIB_DIR="$pypath/lib/@pyLibPrefix@"
```

Comment:
Are you searching for python.pythonForBuild.interpreter ?

Reply:
Unfortunately not.  The `setuptools-rust` package is being build for build platform (host is build) with target to cross platform. The `pythonForBuild` gives me python for build. I need python for target platform.

Reply:
We should make the spliced python available in the passthru probably, like we did now for build only.

Reply:
Actually by using hook that has correct python available solves the primary issue here.